ulohy_04/Caesar_cipher.py

```diff
@@ -0,0 +1,25 @@
+# Vstup - ošetření, že text bude jen ASCII
+# a klíč jen celé a kladné číslo do 26 (máme 26 znaků)
+while True:
+    text = input("Zadej text, který chceš zakódovat: ")
+    if text.isascii():
+        while True:
+            key = input("Jaký klíč chceš použít? ")
+            if key.isdigit() and int(key) > 0 and int(key) % 26 != 0:
+                break
+            else:
+                print('''Klíč musí být celé kladné číslo.
+Násobek 26 nic nezašifruje.''')
```

Comment:
Ten `else` by tam ani nemusel být, v případě, že tam už není nic potom. Bylo by to v tomhle případě trochu čitelnější.

Co se týče používání trojitých uvozovek `'''`, toto není úplně ideální, protože se to vizuálně zalamuje přes začátek dalšího řádku. Používá se to spíše pro opravdu velké bloky nebo pro tzv docs-stringy pro popis chování funkcí.

Lepší by v tomhle případě bylo:
```python
                print('Klíč musí být celé kladné číslo. '
                      'Násobek 26 nic nezašifruje.')
```

Reply:
Tady jsem to chtěla, aby se to vypisovalo na 2 řádky, proto tam byly `'''`. Ale je pravda, že takto je to lepší.

Reply:
Pokud chceš udělat nový řádek ve výpisu, tak to ne zcela souvisí s tím, jak to je zapsané. Nový řádek lze udělat např. takto.

```python
>>> print('Seznam:\n  - papa\n  - lala')
Seznam:
  - papa
  - lala
>>>
```

`\n` - je escape sekvence pro nový řádek.

Podobných sekvencí je více, třeba pro tabulátory `\t` apod.

Viz https://en.wikipedia.org/wiki/Escape_character

Reply:
V tvém případě ideální napsat.

```python
                print('Klíč musí být celé kladné číslo. \n'
                      'Násobek 26 nic nezašifruje.')
```